src/classes/CRLP_Batch_Base_TEST.cls

```diff
@@ -0,0 +1,378 @@
+                userDefinedLastNDaysValueOverride);
+
+        // Create the test data
+        CRLP_RollupContact_TEST.mockRollupCMTValues();
+
+        CRLP_Contact_BATCH contactBatchJob = new CRLP_Contact_BATCH(new List<Id>());
+        contactBatchJob.initializeJobData();
+        contactBatchJob.isScheduledJob = true;
+
+        String generatedWhereClause = contactBatchJob.getWhereClause();
+        System.assert(generatedWhereClause.contains('WHERE'), 'The incremental mode ' +
+                'where clause for the inner join should contain a where filter on the Opportunities.');
+        System.assert(generatedWhereClause.contains(userDefinedLastNDaysFieldOverride));
+        System.assert(generatedWhereClause.contains(String.valueOf(userDefinedLastNDaysValueOverride)));
+    }
+
+    /**
+     * @description Deactivates (in memory) rollups by rollup type and time-bound operation type.
+     * Used to test customized rollup configurations.
+     * @param jobType The RollupType whose time-bound rollups should be deactivated
+     * @param timeBoundOperationType The TimeBoundOperationType to deactivate
+     */
+    private static void deactivateTimeBoundRollups(
+            CRLP_RollupProcessingOptions.RollupType jobType,
+            CRLP_Operation.TimeBoundOperationType timeBoundOperationType) {
+
+        for (Rollup__mdt rollup : CRLP_Rollup_SEL.cachedRollups) {
+            if ((jobType == CRLP_RollupProcessingOptions.RollupType.AccountHardCredit
+                    && isAccountHardCreditRollup(rollup))
+                    || (jobType == CRLP_RollupProcessingOptions.RollupType.ContactHardCredit
+                    && isContactHardCreditRollup(rollup))) {
+                if (rollup.Time_Bound_Operation_Type__c == timeBoundOperationType.name()) {
+                    rollup.Active__c = false;
+                }
+            }
+        }
+    }
+
+    /**
+     * @description Activates or deactivates incremental mode for a RollupType.
+     * Used to test customized rollup configurations.
+     * @param jobType The RollupType whose incremental mode custom setting should be configured
+     * @param isIncrementalMode Whether the incremental mode settings should be activated (true)
+     * or deactivated (false)
+     * @return The configured Customizable_Rollup_Settings__c record
+     */
+    private static Customizable_Rollup_Settings__c configureIncrementalMode(CRLP_RollupProcessingOptions.RollupType jobType, Boolean isIncrementalMode) {
+        Customizable_Rollup_Settings__c settings;
+        if (jobType == CRLP_RollupProcessingOptions.RollupType.AccountHardCredit) {
+            settings = UTIL_CustomSettingsFacade.getRollupSettingsForTests(new Customizable_Rollup_Settings__c (
+                    Customizable_Rollups_Enabled__c = true,
+                    AccountHardCreditNonSkew_Incremental__c = isIncrementalMode
+            ));
+        } else if (jobType == CRLP_RollupProcessingOptions.RollupType.ContactHardCredit) {
```

Comment:
It's hard to see this correctly in github review, but can you add a blank line befe and after the `else if` blocks in all of these methods to break up the lines a bit. It'll make the methods easier to read.